presto-common/src/main/java/com/facebook/presto/common/block/MapBlock.java

```diff
@@ -0,0 +1,303 @@
+
+    private volatile long sizeInBytes;
+
+    /**
+     * Create a map block directly from columnar nulls, keys, values, and offsets into the keys and values.
+     * A null map must have no entries.
+     */
+    public static MapBlock fromKeyValueBlock(
+            int positionCount,
+            Optional<boolean[]> mapIsNull,
+            int[] offsets,
+            Block keyBlock,
+            Block valueBlock)
+    {
+        validateConstructorArguments(0, positionCount, mapIsNull.orElse(null), offsets, keyBlock, valueBlock);
+
+        return createMapBlockInternal(
+                0,
+                positionCount,
+                mapIsNull,
+                offsets,
+                keyBlock,
+                valueBlock,
+                new HashTables(Optional.empty(), positionCount));
+    }
+
+    /**
+     * Create a map block directly without per element validations.
+     * <p>
+     * Internal use by this package and com.facebook.presto.spi.Type only.
+     */
+    public static MapBlock createMapBlockInternal(
+            int startOffset,
+            int positionCount,
+            Optional<boolean[]> mapIsNull,
+            int[] offsets,
+            Block keyBlock,
+            Block valueBlock,
+            HashTables hashTables)
+    {
+        validateConstructorArguments(startOffset, positionCount, mapIsNull.orElse(null), offsets, keyBlock, valueBlock);
+        requireNonNull(hashTables, "hashTables is null");
+        return new MapBlock(
+                startOffset,
+                positionCount,
+                mapIsNull.orElse(null),
+                offsets,
+                keyBlock,
+                valueBlock,
+                hashTables);
+    }
+
+    private static void validateConstructorArguments(
+            int startOffset,
+            int positionCount,
+            @Nullable boolean[] mapIsNull,
+            int[] offsets,
+            Block keyBlock,
+            Block valueBlock)
+    {
+        if (startOffset < 0) {
+            throw new IllegalArgumentException("startOffset is negative");
+        }
+
+        if (positionCount < 0) {
+            throw new IllegalArgumentException("positionCount is negative");
+        }
+
+        if (mapIsNull != null && mapIsNull.length - startOffset < positionCount) {
+            throw new IllegalArgumentException("isNull length is less than positionCount");
+        }
+
+        requireNonNull(offsets, "offsets is null");
+        if (offsets.length - startOffset < positionCount + 1) {
+            throw new IllegalArgumentException("offsets length is less than positionCount");
+        }
+
+        requireNonNull(keyBlock, "keyBlock is null");
+        requireNonNull(valueBlock, "valueBlock is null");
+        if (keyBlock.getPositionCount() != valueBlock.getPositionCount()) {
+            throw new IllegalArgumentException(format("keyBlock and valueBlock has different size: %s %s", keyBlock.getPositionCount(), valueBlock.getPositionCount()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getPositionCount&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 124. Consider formatting this data with &#39;%d&#39; instead.